src/Components/Server/src/ComponentHub.cs

```diff
@@ -127,6 +127,10 @@ public async ValueTask<string> StartCircuit(string baseUri, string uri, string s
                 var store = !string.IsNullOrEmpty(applicationState) ?
                     new ProtectedPrerenderComponentApplicationStore(applicationState, _dataProtectionProvider) :
                     new ProtectedPrerenderComponentApplicationStore(_dataProtectionProvider);
+                var endpoint = Context.GetHttpContext().GetEndpoint();
```

Comment:
@javiercn I'm not sure we can really do this. Doesn't `Context.GetHttpContext()` sometimes return `null`, e.g. if you're on Azure SignalR Service?

Did we have other some way to get endpoint metadata into a SignalR Hub method in the past, e.g., to configure root components from `Startup.cs`?

Reply:
I believe they fixed that, but Brennan can confirm. 

They replicate the context to a degree (that's how things like auth flow)

Reply:
I can't confidently say they fixed it, but from the issue linked, it looks like they did something.

> They replicate the context to a degree (that's how things like auth flow)

Right, and the issue with that is that the endpoint metadata probably isn't copied.

Reply:
@BrennanConroy do you have a recommendation on how we can flow information on to the hub? Is there a transparent way we can do this without explicit knowledge of the signalr service?

(essentially setting it up somewhere)

Reply:
This might be the relevant code, and as you can see, not much is copied
https://github.com/Azure/azure-signalr/blob/39527960fc47a2b8d566cba388ed18f38390dbff/src/Microsoft.Azure.SignalR/ServerConnections/ClientConnectionContext.cs#L272

Reply:
> (essentially setting it up somewhere)

Do you know the information when the service collection is being built? Because you can inject a service into the hub ctor.

Reply:
We wanted to have different info on different hubs based on the endpoint metadata.

If necessary we could limit this to having only one configuration in common to all the hubs but it would be slightly less flexible then.

Reply:
OK, looking at the code, it doesn't appear possible to pass through metadata in general, because there's no guarantee that metadata in general is even serializable.

Since we don't have a known customer requirement to configure different JS root components for different endpoints, I'm going to move this configuration into the services layer (as per @BrennanConroy's suggestion).

If, in the future, some definite need arises to have different JS root components for different endpoints, we could add that as a further feature. We would need to create some way of serializing this information into a claim, I think. I'm sure we could do it but it doesn't seem justified to do it right now.

Reply:
Done